### DISCUSSION TITLE
Issue with building iOS variant

### DISCUSSION THREAD

**Discussion by medonja** [NUMde/compass-numapp-frontend#174] [Q&A]
It seems like I am either missing on something while building the iOS variant, or the new templating system of the frontend has broke something.

When I try to build the app for iOS, I get the following error output:

`% npx react-native run-ios
info Found Xcode workspace "ukbtest.xcworkspace"
info Building (using "xcodebuild -workspace ukbtest.xcworkspace -configuration Debug -scheme ukbtest -destination id=BF1B2A2D-7FE5-436C-9798-813EB7A18B95")
error Failed to build iOS project. We ran "xcodebuild" command but it exited with error code 65. To debug build logs further, consider building your app with Xcode.app, by opening ukbtest.xcworkspace.
Command line invocation:
    /Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild -workspace ukbtest.xcworkspace -configuration Debug -scheme ukbtest -destination id=BF1B2A2D-7FE5-436C-9798-813EB7A18B95
`
...
`ld: warning: directory not found for option '-L/Users/marko/Library/Developer/Xcode/DerivedData/ukbtest-azcdyuznnbsthhaehliurekhecku/Build/Products/Debug-iphonesimulator/DoubleConversion'
ld: warning: directory not found for option '-L/Users/marko/Library/Developer/Xcode/DerivedData/ukbtest-azcdyuznnbsthhaehliurekhecku/Build/Products/Debug-iphonesimulator/FBReactNativeSpec'
ld: warning: directory not found for option '-L/Users/marko/Library/Developer/Xcode/DerivedData/ukbtest-azcdyuznnbsthhaehliurekhecku/Build/Products/Debug-iphonesimulator/FirebaseCore'
ld: warning: directory not found for option '-L/Users/marko/Library/Developer/Xcode/DerivedData/ukbtest-azcdyuznnbsthhaehliurekhecku/Build/Products/Debug-iphonesimulator/FirebaseCoreDiagnostics'
ld: warning: directory not found for option '-L/Users/marko/Library/Developer/Xcode/DerivedData/ukbtest-azcdyuznnbsthhaehliurekhecku/Build/Products/Debug-iphonesimulator/FirebaseInstallations'
ld: warning: directory not found for option '-L/Users/marko/Library/Developer/Xcode/DerivedData/ukbtest-azcdyuznnbsthhaehliurekhecku/Build/Products/Debug-iphonesimulator/FirebaseMessaging'
ld: warning: directory not found for option '-L/Users/marko/Library/Developer/Xcode/DerivedData/ukbtest-azcdyuznnbsthhaehliurekhecku/Build/Products/Debug-iphonesimulator/GoogleDataTransport'
ld: warning: directory not found for option '-L/Users/marko/Library/Developer/Xcode/DerivedData/ukbtest-azcdyuznnbsthhaehliurekhecku/Build/Products/Debug-iphonesimulator/GoogleUtilities'
ld: warning: directory not found for option '-L/Users/marko/Library/Developer/Xcode/DerivedData/ukbtest-azcdyuznnbsthhaehliurekhecku/Build/Products/Debug-iphonesimulator/Permission-Camera'
ld: warning: directory not found for option '-L/Users/marko/Library/Developer/Xcode/DerivedData/ukbtest-azcdyuznnbsthhaehliurekhecku/Build/Products/Debug-iphonesimulator/Permission-Notifications'
ld: warning: directory not found for option '-L/Users/marko/Library/Developer/Xcode/DerivedData/ukbtest-azcdyuznnbsthhaehliurekhecku/Build/Products/Debug-iphonesimulator/PromisesObjC'
ld: warning: directory not found for option '-L/Users/marko/Library/Developer/Xcode/DerivedData/ukbtest-azcdyuznnbsthhaehliurekhecku/Build/Products/Debug-iphonesimulator/RCT-Folly'
ld: warning: directory not found for option '-L/Users/marko/Library/Developer/Xcode/DerivedData/ukbtest-azcdyuznnbsthhaehliurekhecku/Build/Products/Debug-iphonesimulator/RCTTypeSafety'
ld: warning: directory not found for option '-L/Users/marko/Library/Developer/Xcode/DerivedData/ukbtest-azcdyuznnbsthhaehliurekhecku/Build/Products/Debug-iphonesimulator/RNCPicker'
ld: warning: directory not found for option '-L/Users/marko/Library/Developer/Xcode/DerivedData/ukbtest-azcdyuznnbsthhaehliurekhecku/Build/Products/Debug-iphonesimulator/RNDateTimePicker'
ld: warning: directory not found for option '-L/Users/marko/Library/Developer/Xcode/DerivedData/ukbtest-azcdyuznnbsthhaehliurekhecku/Build/Products/Debug-iphonesimulator/RNFBApp'
ld: warning: directory not found for option '-L/Users/marko/Library/Developer/Xcode/DerivedData/ukbtest-azcdyuznnbsthhaehliurekhecku/Build/Products/Debug-iphonesimulator/RNFBMessaging'
ld: warning: directory not found for option '-L/Users/marko/Library/Developer/Xcode/DerivedData/ukbtest-azcdyuznnbsthhaehliurekhecku/Build/Products/Debug-iphonesimulator/RNGestureHandler'
ld: warning: directory not found for option '-L/Users/marko/Library/Developer/Xcode/DerivedData/ukbtest-azcdyuznnbsthhaehliurekhecku/Build/Products/Debug-iphonesimulator/RNLocalize'
ld: warning: directory not found for option '-L/Users/marko/Library/Developer/Xcode/DerivedData/ukbtest-azcdyuznnbsthhaehliurekhecku/Build/Products/Debug-iphonesimulator/RNPermissions'
ld: warning: directory not found for option '-L/Users/marko/Library/Developer/Xcode/DerivedData/ukbtest-azcdyuznnbsthhaehliurekhecku/Build/Products/Debug-iphonesimulator/RNScreens'
ld: warning: directory not found for option '-L/Users/marko/Library/Developer/Xcode/DerivedData/ukbtest-azcdyuznnbsthhaehliurekhecku/Build/Products/Debug-iphonesimulator/RNVectorIcons'
ld: warning: directory not found for option '-L/Users/marko/Library/Developer/Xcode/DerivedData/ukbtest-azcdyuznnbsthhaehliurekhecku/Build/Products/Debug-iphonesimulator/React-Core'
ld: warning: directory not found for option '-L/Users/marko/Library/Developer/Xcode/DerivedData/ukbtest-azcdyuznnbsthhaehliurekhecku/Build/Products/Debug-iphonesimulator/React-CoreModules'
ld: warning: directory not found for option '-L/Users/marko/Library/Developer/Xcode/DerivedData/ukbtest-azcdyuznnbsthhaehliurekhecku/Build/Products/Debug-iphonesimulator/React-RCTAnimation'
ld: warning: directory not found for option '-L/Users/marko/Library/Developer/Xcode/DerivedData/ukbtest-azcdyuznnbsthhaehliurekhecku/Build/Products/Debug-iphonesimulator/React-RCTBlob'
ld: warning: directory not found for option '-L/Users/marko/Library/Developer/Xcode/DerivedData/ukbtest-azcdyuznnbsthhaehliurekhecku/Build/Products/Debug-iphonesimulator/React-RCTImage'
ld: warning: directory not found for option '-L/Users/marko/Library/Developer/Xcode/DerivedData/ukbtest-azcdyuznnbsthhaehliurekhecku/Build/Products/Debug-iphonesimulator/React-RCTLinking'
ld: warning: directory not found for option '-L/Users/marko/Library/Developer/Xcode/DerivedData/ukbtest-azcdyuznnbsthhaehliurekhecku/Build/Products/Debug-iphonesimulator/React-RCTNetwork'
ld: warning: directory not found for option '-L/Users/marko/Library/Developer/Xcode/DerivedData/ukbtest-azcdyuznnbsthhaehliurekhecku/Build/Products/Debug-iphonesimulator/React-RCTSettings'
ld: warning: directory not found for option '-L/Users/marko/Library/Developer/Xcode/DerivedData/ukbtest-azcdyuznnbsthhaehliurekhecku/Build/Products/Debug-iphonesimulator/React-RCTText'
ld: warning: directory not found for option '-L/Users/marko/Library/Developer/Xcode/DerivedData/ukbtest-azcdyuznnbsthhaehliurekhecku/Build/Products/Debug-iphonesimulator/React-RCTVibration'
ld: warning: directory not found for option '-L/Users/marko/Library/Developer/Xcode/DerivedData/ukbtest-azcdyuznnbsthhaehliurekhecku/Build/Products/Debug-iphonesimulator/React-cxxreact'
ld: warning: directory not found for option '-L/Users/marko/Library/Developer/Xcode/DerivedData/ukbtest-azcdyuznnbsthhaehliurekhecku/Build/Products/Debug-iphonesimulator/React-jsi'
ld: warning: directory not found for option '-L/Users/marko/Library/Developer/Xcode/DerivedData/ukbtest-azcdyuznnbsthhaehliurekhecku/Build/Products/Debug-iphonesimulator/React-jsiexecutor'
ld: warning: directory not found for option '-L/Users/marko/Library/Developer/Xcode/DerivedData/ukbtest-azcdyuznnbsthhaehliurekhecku/Build/Products/Debug-iphonesimulator/React-jsinspector'
ld: warning: directory not found for option '-L/Users/marko/Library/Developer/Xcode/DerivedData/ukbtest-azcdyuznnbsthhaehliurekhecku/Build/Products/Debug-iphonesimulator/React-logger'
ld: warning: directory not found for option '-L/Users/marko/Library/Developer/Xcode/DerivedData/ukbtest-azcdyuznnbsthhaehliurekhecku/Build/Products/Debug-iphonesimulator/React-perflogger'
ld: warning: directory not found for option '-L/Users/marko/Library/Developer/Xcode/DerivedData/ukbtest-azcdyuznnbsthhaehliurekhecku/Build/Products/Debug-iphonesimulator/ReactCommon'
ld: warning: directory not found for option '-L/Users/marko/Library/Developer/Xcode/DerivedData/ukbtest-azcdyuznnbsthhaehliurekhecku/Build/Products/Debug-iphonesimulator/Yoga'
ld: warning: directory not found for option '-L/Users/marko/Library/Developer/Xcode/DerivedData/ukbtest-azcdyuznnbsthhaehliurekhecku/Build/Products/Debug-iphonesimulator/fmt'
ld: warning: directory not found for option '-L/Users/marko/Library/Developer/Xcode/DerivedData/ukbtest-azcdyuznnbsthhaehliurekhecku/Build/Products/Debug-iphonesimulator/glog'
ld: warning: directory not found for option '-L/Users/marko/Library/Developer/Xcode/DerivedData/ukbtest-azcdyuznnbsthhaehliurekhecku/Build/Products/Debug-iphonesimulator/nanopb'
ld: warning: directory not found for option '-L/Users/marko/Library/Developer/Xcode/DerivedData/ukbtest-azcdyuznnbsthhaehliurekhecku/Build/Products/Debug-iphonesimulator/react-native-camera'
ld: warning: directory not found for option '-L/Users/marko/Library/Developer/Xcode/DerivedData/ukbtest-azcdyuznnbsthhaehliurekhecku/Build/Products/Debug-iphonesimulator/react-native-encrypted-storage'
ld: warning: directory not found for option '-L/Users/marko/Library/Developer/Xcode/DerivedData/ukbtest-azcdyuznnbsthhaehliurekhecku/Build/Products/Debug-iphonesimulator/react-native-safe-area-context'
ld: warning: directory not found for option '-L/Users/marko/Library/Developer/Xcode/DerivedData/ukbtest-azcdyuznnbsthhaehliurekhecku/Build/Products/Debug-iphonesimulator/react-native-slider'
ld: warning: directory not found for option '-L/Users/marko/Library/Developer/Xcode/DerivedData/ukbtest-azcdyuznnbsthhaehliurekhecku/Build/Products/Debug-iphonesimulator/react-native-splash-screen'
ld: warning: directory not found for option '-L/Users/marko/Library/Developer/Xcode/DerivedData/ukbtest-azcdyuznnbsthhaehliurekhecku/Build/Products/Debug-iphonesimulator/react-native-webview'
ld: library not found for -lDoubleConversion
clang: error: linker command failed with exit code 1 (use -v to see invocation)

2022-06-09 16:40:59.976 xcodebuild[4006:40124] Requested but did not find extension point with identifier Xcode.IDEKit.ExtensionSentinelHostApplications for extension Xcode.DebuggerFoundation.AppExtensionHosts.watchOS of plug-in com.apple.dt.IDEWatchSupportCore
2022-06-09 16:40:59.976 xcodebuild[4006:40124] Requested but did not find extension point with identifier Xcode.IDEKit.ExtensionPointIdentifierToBundleIdentifier for extension Xcode.DebuggerFoundation.AppExtensionToBundleIdentifierMap.watchOS of plug-in com.apple.dt.IDEWatchSupportCore
** BUILD FAILED **
`

This problem however, did not occur with a previous commit (I tried to build the commit 368a0c8a553 for iOS, and have been successful).

Can you help me out with this one?

**Comment by mbastian93** [comment]
Did you cd into the ios folder and run pod install?

**Reply by medonja** [reply]
Yes, even a couple of times...

**Reply by mbastian93** [reply]
Maybe open the *.xcworkspace with Xcode, clean the project and try to build it from there?

**Reply by medonja** [reply]
Yes, I have tried this as well and it still did not work. It seems also that most stackoverflow posts regarding this issue recommend the same steps, but for me they do not work. What is weird though, is that I tried just for the sake of comparison, to do a clean checkout of the aformentioned commit as well as the template of v1.0.2 - and it only worked with the previous commit, but not with v1.0.2.

**Reply by mbastian93** [reply]
Could you maybe try to setup a new project and use the repo url without version (https://github.com/NUMde/compass-numapp-frontend) and see if that works for you?

**Reply by medonja** [reply]
Hi, I just tried this, and the error returned is the same, as with the template version. Again, when I try to build a more previous commit (e.g. 368a0c8) the build runs successfully for iOS.

**Reply by mbastian93** [reply]
I'm afraid i can't reproduce the error...

Maybe [this thread](https://github.com/facebook/react-native/issues/25240) contains some useful advice

**Reply by medonja** [reply]
I know now what is causing the problem. You are probably using an Intel-based Mac, right? I am working on an M1 Mac. So, I have tried running XCode using Rosetta, et voila, no more build problems! Now, I don't know what exactly is causing the issue, but perhaps lDoubleConversion library is not available for M1 yet, or something in that direction? That would also explain why some users report solving the issue successfully by following the steps explained (open workspace, do pod install etc...) while others were not successful by following the instructions and remain frustrated.
